src/variety/snake.js

```diff
@@ -0,0 +1,409 @@
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					this.inputcell();
+				}
+			} else if (this.puzzle.editmode) {
+				if (this.mousestart) {
+					if (!this.inputqnum_excell()) {
+						this.inputqnum();
+					}
+				}
+			}
+		},
+
+		inputqnum_excell: function() {
+			var excell = this.getcell_excell();
+			if (excell.isnull || excell.group !== "excell") {
+				return false;
+			}
+
+			if (excell !== this.cursor.getex()) {
+				this.setcursor(excell);
+			} else {
+				this.inputqnum_main(excell);
+			}
+			return true;
+		}
+	},
+
+	KeyEvent: {
+		enablemake: true,
+
+		keyinput: function(ca) {
+			if (!this.key_inputexcell(ca)) {
+				this.key_inputqnum(ca);
+			}
+		},
+
+		key_inputexcell: function(ca) {
+			var excell = this.cursor.getex(),
+				qn = excell.qnum;
+
+			if (excell.isnull) {
+				return false;
+			}
+			var max = excell.getmaxnum();
+
+			if ("0" <= ca && ca <= "9") {
+				var num = +ca;
+
+				if (qn <= 0 || this.prev !== excell) {
+					if (num <= max) {
+						excell.setQnum(num);
+					}
+				} else {
+					if (qn * 10 + num <= max) {
+						excell.setQnum(qn * 10 + num);
+					} else if (num <= max) {
+						excell.setQnum(num);
+					}
+				}
+			} else if (ca === " " || ca === "-") {
+				excell.setQnum(-1);
+			} else {
+				return true;
+			}
+
+			this.prev = excell;
+			this.cursor.draw();
+			return true;
+		}
+	},
+
+	TargetCursor: {
+		initCursor: function() {
+			this.init(-1, -1);
+		}
+	},
+
+	Cell: {
+		numberAsObject: true,
+		disInputHatena: true,
+		maxnum: 2,
+
+		prehook: {
+			qsub: function() {
+				return this.qnum !== -1;
+			}
+		},
+
+		shades: function() {
+			return this.countDir4Cell(function(adj) {
+				return adj.isShade();
+			});
+		}
+	},
+
+	ExCell: {
+		disInputHatena: true,
+
+		maxnum: function() {
+			var bx = this.bx,
+				by = this.by;
+			if (bx === -1 && by === -1) {
+				return 0;
+			}
+			return by === -1 ? this.board.rows : this.board.cols;
+		},
+		minnum: 0
+	},
+
+	Board: {
+		hasexcell: 1,
+
+		cols: 8,
+		rows: 8
+	},
+
+	AreaShadeGraph: {
+		enabled: true
+	},
+
+	BoardExec: {
+		adjustBoardData: function(key, d) {
+			this.adjustExCellTopLeft_1(key, d);
+		},
+		adjustBoardData2: function(key, d) {
+			this.adjustExCellTopLeft_2(key, d);
+		}
+	},
+
+	Graphic: {
+		enablebcolor: true,
+
+		shadecolor: "rgb(80, 80, 80)",
+
+		circlefillcolor_func: "qnum2",
+		getCircleStrokeColor: function(cell) {
+			return cell.qnum > 0 ? this.quescolor : null;
+		},
+
+		paint: function() {
+			this.drawBGCells();
+			this.drawDotCells();
+
+			this.drawShadedCells();
+			this.drawGrid();
+
+			this.drawNumbersExCell();
+			this.drawCircles();
+
+			this.drawChassis();
+
+			this.drawTarget();
+		}
+	},
+
+	Encode: {
+		decodePzpr: function(type) {
+			this.decodeCircle();
+			this.outbstr = this.outbstr.substr(1);
+			this.decodeNumber16ExCell();
+		},
+		encodePzpr: function(type) {
+			this.encodeCircle();
+			this.outbstr += "/";
```

Comment:
Any particular reason to add this separator? Other types don't tend to do that. (If we were to start something like this, the decoder should make use of it, i.e., split on `/` and then decode both parts separately.)

Reply:
Aquarium needed the extra slash (for reasons I can't recall), but Snake apparently doesn't. I've removed it here.